tck/metrics/src/main/java/org/eclipse/microprofile/telemetry/metrics/tck/jvm/MetricsReader.java

```diff
@@ -21,36 +21,65 @@
  **********************************************************************/
 package org.eclipse.microprofile.telemetry.metrics.tck.jvm;
 
-import java.io.BufferedReader;
-import java.io.FileReader;
-import java.io.IOException;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.commons.io.input.Tailer;
+import org.apache.commons.io.input.TailerListenerAdapter;
+import org.awaitility.Awaitility;
+import org.testng.Assert;
 
 public class MetricsReader {
 
     private static final String logFilePath = System.getProperty("mptelemetry.tck.log.file.path");
 
-    public static boolean checkMessage(String metricName, String metricDescription, String metricUnit,
-            String metricType)
-            throws IOException {
-        try {
-            Thread.sleep(5000);
-            try {
-                BufferedReader reader = new BufferedReader(new FileReader(logFilePath));
-                String line;
-                while ((line = reader.readLine()) != null) {
-                    if (line.contains(
-                            "name=" + metricName + ", description=" + metricDescription + ", unit=" + metricUnit
-                                    + ", type=" + metricType)) {
-
-                        return true;
-                    }
-                }
-                return false;
-            } catch (IOException e) {
-                return false;
+    public static void checkMessage(String metricName, String metricDescription, String metricUnit,
+            String metricType) {
```

Comment:
This method should probably have Javadoc explaining exactly what it looks fot.

Since you've changed it from returning a result to asserting, it should probably be renamed to `assertMessage`.